### PR TITLE
Validate channel count in rgb_to_grayscale and RandomColorDegeneration

### LINKED ISSUE #22536: RandomColorDegeneration fails on eager tensor inputs with matrix size-incompatible error
## Summary
`keras.layers.RandomColorDegeneration` raises a TensorFlow `Matrix size-incompatible` error when called on an eager tensor, but works on a `keras.Input` placeholder.

## Reproduction
```python
import keras
import numpy as np
import tensorflow as tf

x = tf.constant(np.random.rand(2, 224, 224, 3))
layer = tf.keras.layers.RandomColorDegeneration(
    factor=[0.2, 0.8],
    value_range=(0, 255),
    data_format="channels_first",
    seed=42,
)
try:
    print(layer(x))
except Exception as e:
    print(f"Error: {e}")
print("\n" + "=" * 80 + "\n")
sym_x = keras.Input(shape=(224, 224, 3))
try:
    sym_y = layer(sym_x)
    model = keras.Model(inputs=sym_x, outputs=sym_y)
    print(model.summary())
except Exception as e:
    print(f"Error: {e}")
```
output:

```
W0000 00:00:1774513553.655934  150308 local_rendezvous.cc:412] Local rendezvous is aborting with status: INVALID_ARGUMENT: Matrix size-incompatible: In[0]: [1344,224], In[1]: [3,1]
Error: Exception encountered when calling RandomColorDegeneration.call().

{{function_node __wrapped____MklMatMul_device_/job:localhost/replica:0/task:0/device:CPU:0}} Matrix size-incompatible: In[0]: [1344,224], In[1]: [3,1] [Op:MatMul] name: 

Arguments received by RandomColorDegeneration.call():
  • data=tf.Tensor(shape=(2, 224, 224, 3), dtype=float32)
  • training=True

================================================================================

Model: "functional"
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ Layer (type)                         ┃ Output Shape                ┃         Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ input_layer (InputLayer)             │ (None, 224, 224, 3)         │               0 │
├──────────────────────────────────────┼─────────────────────────────┼─────────────────┤
│ random_color_degeneration            │ (None, 224, 224, 3)         │               0 │
│ (RandomColorDegeneration)            │                             │                 │
└──────────────────────────────────────┴─────────────────────────────┴─────────────────┘
 Total params: 0 (0.00 B)
 Trainable params: 0 (0.00 B)
 Non-trainable params: 0 (0.00 B)
None
```

### PR DESCRIPTION
**Pull request by rstar327**
## Description

Fixes #22536.

`keras.ops.image.rgb_to_grayscale` and `keras.layers.RandomColorDegeneration` previously failed inconsistently when the channel axis size did not match the expected RGB count (3 for RGB or 1 for grayscale):

- **Eager** execution crashed with a confusing `Matrix size-incompatible` error (TF/JAX/NumPy backends).
- **Symbolic** execution silently succeeded, even though the configuration was invalid.

This PR adds explicit channel validation:

- `RGBToGrayscale.compute_output_spec` now validates the channel axis size (matches `RGBToHSV` behavior).
- `rgb_to_grayscale` in TF, JAX, and NumPy backends now validates the channel axis size and treats single-channel input as a no-op (matches existing Torch and OpenVINO behavior).
- `RandomColorDegeneration.compute_output_shape` now validates the channel axis size so the symbolic path raises a clear error before runtime.
- Added a unit test covering the invalid channel case.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

**Comment by rstar327** [comment]
Done

**Review by hertschuh — APPROVED** [review]
Thanks for the fixes!
